Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java

```diff
@@ -323,64 +325,107 @@ public void stopWebApp(String sid, String appid) throws IOException {
     }
 
     /**
-     * List Web Apps by Subscription ID.
+     * List all the Web Apps in selected subscriptions.
+     *
+     * @param force flag indicating whether force to fetch most updated data from server
+     * @return list of Web App
      */
-    public List<ResourceEx<WebApp>> listWebAppsOnWindowsBySubscriptionId(String sid, boolean force) {
-        if (!force && subscriptionIdToWebAppsOnWindowsMap.containsKey(sid)) {
-            return subscriptionIdToWebAppsOnWindowsMap.get(sid);
-        }
-        List<ResourceEx<WebApp>> webAppList = new ArrayList<>();
-        try {
-            Azure azure = AuthMethodManager.getInstance().getAzureClient(sid);
-            for (WebApp webApp : azure.webApps().list()) {
-                if (webApp.operatingSystem().equals(OperatingSystem.WINDOWS)) {
-                    webAppList.add(new ResourceEx<>(webApp, sid));
-                }
+    public List<ResourceEx<WebApp>> listAllWebApps(boolean force) {
```

Comment:
Suggest to leverage `listWebApps(subscriptionId, force)`. Then the logic here would be clear: for each selected sid, list web apps, merge them to a list, return.